lib/email_processor.rb

```diff
@@ -3,24 +3,42 @@
 class EmailProcessor
   def initialize(email)
     @email = email
+    @content, @course, @from_address, @owner, @reference_id, @sender = nil
   end
 
   def process
+    define_owner
+    define_sender
+    define_content_and_reference_id
+    dispense_or_thread_ticket
+  end
+
+  def retrieve_forwarder_email
+    raw_body = @email.raw_body
+    expression = /\b[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{0,63}\b/i
+    raw_body.scan(expression).first
```

Comment:
This will still scan the entire body for all matches... I think `raw_body.match(expression)` (or, if you want to get fancy, `raw_body[expression]`) would be the more efficient way, although it's trivial.